OSDOCS-11208 Management Console bugs for 4.17 RNs (#82467)
openshift/openshift-docs: release_notes/ocp-4-17-release-notes.adoc

[id="ocp-4-17-management-console-bug-fixes_{context}"]
==== Management Console

> * Previously, a warning was not provided when you were on a Google Cloud Platform (GCP) cluster that supports GCP's Workload Identity Management, and that the Operator supports it. With this release, logic was added support for GCP Workload Identity and Federated Identity Operator installs, so now you are alerted when you are on a GCP cluster. (link:https://issues.redhat.com/browse/OCPBUGS-38591[*OCPBUGS-38591*])

Reviewer: ```suggestion
* Previously, a warning was not provided when you were on a Google Cloud Platform (GCP) cluster that supports GCP's Workload Identity Management and that the Operator supports it. With this release, logic was added to support GCP Workload Identity and Federated Identity Operator installs, so now you are alerted when you are on a GCP cluster. (link:https://issues.redhat.com/browse/OCPBUGS-38591[*OCPBUGS-38591*])
```

* Previously, the *Cluster overview* page included a `View all steps in documentation` link that resulted in a 404 error for ROSA and OSD clusters. With this update, the link does not appear for ROSA and OSD clusters. (link:https://issues.redhat.com/browse/OCPBUGS-37054[*OCPBUGS-37054*])

* Previously, dynamic plugins using PatternFly 4 were referencing variables that are not available in {product-title} 4.15 and later. This was causing contrast issues for ACM in dark mode. With this update, older chart styles are now available to support PatternFly 4 charts used by dynamic plugins. (link:https://issues.redhat.com/browse/OCPBUGS-36816[*OCPBUGS-36816*])

* Previously, when the `Display Admission Webhook` warning implementation presented issues with some incorrect code. With this update, the unnecessary warning message has been removed. (link:https://issues.redhat.com/browse/OCPBUGS-35940[*OCPBUGS-35940*])

> * Previously, the global sync lock that applied to all HTTP-serving spawned gorotines with a sync lock that is specific to each of the refresh tokens. With this update, the global refresh sync lock on a cluster with an external OIDC environment was replaced with per-refresh token one. As a result, refresh token performance is improved by 30 to 50%. (link:https://issues.redhat.com/browse/OCPBUGS-35080[*OCPBUGS-35080*])

Reviewer: Should this be: "Previously, the global sync lock that applied to all HTTP-servers..."?

Also, just want to make sure _gorotines_ is spelled correctly. Should it be _goroutines_?

Reviewer: ```suggestion
* Previously, the global sync lock that applied to all HTTP-serving spawned gorotines with a sync lock that is specific to each of the refresh tokens. With this update, the global refresh sync lock on a cluster with an external OIDC environment was replaced with a per-refresh token one. As a result, refresh token performance is improved by 30 to 50%. (link:https://issues.redhat.com/browse/OCPBUGS-35080[*OCPBUGS-35080*])
```

Reviewer: Last thing, I'm not sure what "one" refers to in "per-refresh token one." Do you mean "per-refresh token sync"? If so, you could reword this sentence something like, "With this update, the global refresh sync lock on a cluster with an external OIDC environment was replaced with a sync that refreshes for each token."

> * Previously, a warning was not displayed for `minAvailable` in `PodDisruptionBudget` create and edit form. With this update, code logic for displaying `minAvailable` warning was added, and the `minAvailable` warning is displayed if violated. (link:https://issues.redhat.com/browse/OCPBUGS-34937[*OCPBUGS-34937*])

Reviewer: ```suggestion
* Previously, a warning was not displayed for the `minAvailable` warning in `PodDisruptionBudget` create and edit form. With this update, code logic for displaying the `minAvailable` warning was added, and the `minAvailable` warning is displayed if violated. (link:https://issues.redhat.com/browse/OCPBUGS-34937[*OCPBUGS-34937*])
```

* Previously, the *OperandDetails* page displayed information for the first CRD that matched by name. After this fix, the *OperandDetails* page displays information for the CRD that matches by name and the version of the operand. (link:https://issues.redhat.com/browse/OCPBUGS-34901[*OCPBUGS-34901*])

> * Starting with 4.16 release, Console provided PatternFly 4.x shared modules are deprecated and will be removed in the future. console plugins are encouraged to upgrade to PatternFly 5. Also starting with 4.16 release, React Router v5 shared modules that were provided are now deprecated and will be removed in the future.

Reviewer: ```suggestion
* Starting with the 4.16 release, console-provided PatternFly 4.x shared modules are deprecated and will be removed in the future. You are encouraged to upgrade your console plugins to PatternFly 5. Also starting with the 4.16 release, React Router v5 shared modules that were provided are now deprecated and will be removed in the future.
```

Author: This doesn't actually belong here so I'm removing it.

You are encouraged to upgrade your dynamic plugins to React Router v6 with the `react-router-dom-v5-compat` module. (link:https://issues.redhat.com/browse/OCPBUGS-34538[*OCPBUGS-34538*])

* Previously, one inactive or idle browser tab caused session expiration for all other tabs. With this change, activity in any tab will prevent session expiration even if there is one inactive or idle browser tab. (link:https://issues.redhat.com/browse/OCPBUGS-34387[*OCPBUGS-34387*])

> * Previously, when the `Display Admission Webhook` warning implementation presented issues with some incorrect code. With this release, the unnecessary warning message has been removed. (link:https://issues.redhat.com/browse/OCPBUGS-35940[*OCPBUGS-34316*])

Reviewer: ```suggestion
* Previously, the `Display Admission Webhook` warning implementation presented issues with some incorrect code. With this release, the unnecessary warning message has been removed. (link:https://issues.redhat.com/browse/OCPBUGS-35940[*OCPBUGS-34316*])
```

* Previously, the *Debug container* link was not displayed for pods with a `Completed` status. With this change, the link now appears. (link:https://issues.redhat.com/browse/OCPBUGS-33631[*OCPBUGS-33631*])

* Previously, the {product-title} web console did not show `filesystem` metrics on the *Nodes list* page due to incorrect prom query. With this update, `filesystem` metrics are correctly displayed. (link:https://issues.redhat.com/browse/OCPBUGS-33136[*OCPBUGS-33136*])

* Previously, pseudolocalization was not working due to a configuration issue. After this fix, pseudolocalization works again. (link:https://issues.redhat.com/browse/OCPBUGS-30218[*OCPBUGS-30218*])

* Previously, console pods would crash loop if the `--user-auth` flag was set to `disabled`. With this update, the console backend properly handles this value. (link:https://issues.redhat.com/browse/OCPBUGS-29510[*OCPBUGS-29510*])

> * Previously, utilization cards displayed a `limit` incorrectly implying a relationship between capacity & limits. With this update, the position of `limit` was changed and the wording updated. (link:https://issues.redhat.com/browse/OCPBUGS-23332[*OCPBUGS-23332*])

Reviewer: ```suggestion
* Previously, utilization cards displayed a `limit` that incorrectly implied a relationship between capacity and limits. With this update, the position of `limit` was changed and the wording updated. (link:https://issues.redhat.com/browse/OCPBUGS-23332[*OCPBUGS-23332*])
```

> * Previously, in some edge cases, the wrong resource could be fetched when using websockets to watch a namespaced resource without providing a namespace. With this update, a validation to the resource watch logic was added to prevent the websocket request, and log an error under this condition. (link:https://issues.redhat.com/browse/OCPBUGS-19855[*OCPBUGS-19855*])

Reviewer: ```suggestion
* Previously, in some edge cases, the wrong resource could be fetched when using websockets to watch a namespaced resource without providing a namespace. With this update, a validation to the resource watch logic was added to prevent the websocket request and log an error under this condition. (link:https://issues.redhat.com/browse/OCPBUGS-19855[*OCPBUGS-19855*])
```

> * Previously, perspective switching was not properly handled. With this update, perspectives passed with URL search parameters or plugin route page extensions now correctly switch the perspective and retain the correct URL path. (link:https://issues.redhat.com/browse/OCPBUGS-19048[*OCPBUGS-19048*])

Reviewer: ```suggestion
* Previously, perspective switching was not properly handled. With this update, perspectives that are passed with URL search parameters or plugin route page extensions now correctly switch the perspective and retain the correct URL path. (link:https://issues.redhat.com/browse/OCPBUGS-19048[*OCPBUGS-19048*])
```

[id="ocp-4-17-management-console-bug-fixes_{context}"]
==== Management Console

* Previously, a warning was not provided when you were on a Google Cloud Platform (GCP) cluster that supports GCP's Workload Identity Management and that the Operator supports it. With this release, logic was added to support GCP Workload Identity and Federated Identity Operator installs, so now you are alerted when you are on a GCP cluster. (link:https://issues.redhat.com/browse/OCPBUGS-38591[*OCPBUGS-38591*])

* Previously, the version number text in the *Updates* graph on the *Cluster Settings* page appeared as black text on a dark background when using Firefox in dark mode. With this update, the text appears as white text. (link:https://issues.redhat.com/browse/OCPBUGS-38427[*OCPBUGS-38427*])

> * Previously, the *Cluster overview* page included a `View all steps in documentation` link that resulted in a 404 error for ROSA and OSD clusters. With this update, the link does not appear for ROSA and OSD clusters. (link:https://issues.redhat.com/browse/OCPBUGS-37054[*OCPBUGS-37054*])

Reviewer 2: ok yeah this probably should be dropped
```suggestion
```

> * Previously, the *Cluster overview* page included a `View all steps in documentation` link that resulted in a 404 error for ROSA and OSD clusters. With this update, the link does not appear for ROSA and OSD clusters. (link:https://issues.redhat.com/browse/OCPBUGS-37054[*OCPBUGS-37054*])

Author: Thanks! Good catch!

* Previously, dynamic plugins using PatternFly 4 were referencing variables that are not available in {product-title} 4.15 and later. This was causing contrast issues for ACM in dark mode. With this update, older chart styles are now available to support PatternFly 4 charts used by dynamic plugins. (link:https://issues.redhat.com/browse/OCPBUGS-36816[*OCPBUGS-36816*])

* Previously, when the `Display Admission Webhook` warning implementation presented issues with some incorrect code. With this update, the unnecessary warning message has been removed. (link:https://issues.redhat.com/browse/OCPBUGS-35940[*OCPBUGS-35940*])

> * Previously, the global sync lock that applied to all HTTP-servers spawned goroutines with a sync lock that is specific to each of the refresh tokens. With this release, the global refresh sync lock on a cluster with an external OIDC environment was replaced with a sync that refreshes for each token. As a result, refresh token performance is improved by 30 to 50%. (link:https://issues.redhat.com/browse/OCPBUGS-35080[*OCPBUGS-35080*])

Reviewer 2: ```suggestion
* Previously, the global sync lock that applied to all HTTP servers spawned goroutines with a sync lock that is specific to each of the refresh tokens. With this release, the global refresh sync lock on a cluster with an external OIDC environment was replaced with a sync that refreshes for each token. As a result, refresh token performance is improved by 30% to 50%. (link:https://issues.redhat.com/browse/OCPBUGS-35080[*OCPBUGS-35080*])
```

* Previously, a warning was not displayed for the `minAvailable` warning in `PodDisruptionBudget` create and edit form. With this update, code logic for displaying the `minAvailable` warning was added, and the `minAvailable` warning is displayed if violated. (link:https://issues.redhat.com/browse/OCPBUGS-34937[*OCPBUGS-34937*])

* Previously, the *OperandDetails* page displayed information for the first CRD that matched by name. After this fix, the *OperandDetails* page displays information for the CRD that matches by name and the version of the operand. (link:https://issues.redhat.com/browse/OCPBUGS-34901[*OCPBUGS-34901*])

* Previously, one inactive or idle browser tab caused session expiration for all other tabs. With this change, activity in any tab will prevent session expiration even if there is one inactive or idle browser tab. (link:https://issues.redhat.com/browse/OCPBUGS-34387[*OCPBUGS-34387*])

* Previously, the `Display Admission Webhook` warning implementation presented issues with some incorrect code. With this release, the unnecessary warning message has been removed. (link:https://issues.redhat.com/browse/OCPBUGS-35940[*OCPBUGS-34316*])

> * Previously, select `textareas` were not resizable. With this update, you are now able to resize `textareas`. (link:https://issues.redhat.com/browse/OCPBUGS-34200[*OCPBUGS-34200*])

Reviewer 2: ```suggestion
* Previously, text areas were not resizable. With this update, you are now able to resize text areas. (link:https://issues.redhat.com/browse/OCPBUGS-34200[*OCPBUGS-34200*])
```

> * Previously, the console Operator was not able to tolerate the absence of the ingress capability. With this update, the console Operator configuration API has been enhanced with the possibility to add alternative ingress for the environments where the ingress cluster capability is disabled. (link:https://issues.redhat.com/browse/OCPBUGS-33787[OCPBUGS-33787*])

Reviewer 2: ```suggestion
* Previously, the Console Operator was not able to tolerate the absence of the ingress capability. With this update, the Console Operator configuration API has been enhanced with the possibility to add alternative ingress for the environments where the ingress cluster capability is disabled. (link:https://issues.redhat.com/browse/OCPBUGS-33787[OCPBUGS-33787*])
```

* Previously, the *Debug container* link was not displayed for pods with a `Completed` status. With this change, the link now appears. (link:https://issues.redhat.com/browse/OCPBUGS-33631[*OCPBUGS-33631*])

> * Previously, the {product-title} web console did not show `filesystem` metrics on the *Nodes list* page due to incorrect prom query. With this update, `filesystem` metrics are correctly displayed. (link:https://issues.redhat.com/browse/OCPBUGS-33136[*OCPBUGS-33136*])

Reviewer 2: Is "prom" Prometheus maybe?
```suggestion
* Previously, the {product-title} web console did not show `filesystem` metrics on the *Nodes list* page due to incorrect Prometheus query. With this update, `filesystem` metrics are correctly displayed. (link:https://issues.redhat.com/browse/OCPBUGS-33136[*OCPBUGS-33136*])
```
